### PR TITLE
Allow configuration of PVC size from ConfigMap

### PR DESCRIPTION
**Pull request by carlossg**
# Changes

Add a ConfigMap for PVC configuration, particularly the size of the PVC

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

PVCs are hardcoded to 5Gi and that should be configurable.
For instance Alibaba Container Service does not allow volumes smaller than 20Gi

Add a `config-artifact-pvc` ConfigMap that can be used to store such configuration

# Release Notes

```
Add a `config-artifact-pvc` ConfigMap that can be used to store the created PVC size
```

### CODE REVIEW

**Comment by carlossg** [comment]
This is a rough PR from my lack of Golang skills :(

File: docs/install.md [hunk 1]
```diff
@@ -104,8 +104,13 @@ Pipelines need a way to share resources between tasks. The alternatives are a
 [Persistent volume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)
 or a [GCS storage bucket](https://cloud.google.com/storage/)
 
-The PVC option does not require any configuration, but the GCS storage bucket
-can be configured using a ConfigMap with the name `config-artifact-bucket` with
+The PVC option can be configured using a ConfigMap with the name
```
Comment (abayer) [inline docs/install.md]: I'd suggest putting something in here about the default size.

**Comment by carlossg** [comment]
`Google Compute Engine does not have enough resources available to fulfill request: us-central1-a.`

😭

File: pkg/artifacts/artifact_storage_test.go [hunk 2]
```diff
@@ -285,3 +285,41 @@ func TestGetArtifactStorageWithoutConfigMap(t *testing.T) {
...
+
+func TestGetArtifactStorageWithPvcConfigMap(t *testing.T) {
+	logger := logtesting.TestLogger(t)
+	prName := "pipelineruntest"
+	for _, c := range []struct {
+		desc                    string
+		configMap               *corev1.ConfigMap
+		expectedArtifactStorage ArtifactStorageInterface
+	}{{
+		desc: "valid bucket",
```
Comment (afrittoli) [inline pkg/artifacts/artifact_storage_test.go]: NIT: valid PVC

File: pkg/artifacts/artifact_storage_test.go [hunk 3]
```diff
@@ -285,3 +285,41 @@ func TestGetArtifactStorageWithoutConfigMap(t *testing.T) {
...
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      PvcConfigName,
+			},
+			Data: map[string]string{
```
Comment (afrittoli) [inline pkg/artifacts/artifact_storage_test.go]: We don't really test that the pvc size that will be requested is the right size (10Gi), we only check that no error is triggered when the config map is loaded. Perhaps we could "mock" the `getPVCSpec` method and verify that it's invoked with the expected values?

**Comment by carlossg** [comment]
I have made `ArtifactPVC` include the pvc for testing. I don't know if it is great as the PVC is only created during initialization, not in `GetArtifactStorage`, and so `GetArtifactStorage` returns an `ArtifactPVC` with nil PVC
I wasn't able to test expectations, ie. check that the mock k8sclient is called with POST `PersistentVolumeClaim` and get the parameter.

File: pkg/artifacts/artifact_storage_test.go (hunk 3 above)
Comment (carlossg) [reply pkg/artifacts/artifact_storage_test.go]: added test for size

**Comment by bobcatfish** [comment]
> I think this could go ahead like this and we could improve test coverage later.

only if we definitely plan to improve it in a follow-up PR! otherwise id definitely rather err on the side of having it all in one PR (tests + docs + code)

**Comment by carlossg** [comment]
I've added some tests before it was merged

**Comment by bobcatfish** [comment]
> I've added some tests before it was merged

phew, close one :D :heart:

**Pull request merged**